### PR TITLE
fix: Legacy bugs (styles, site plugin error, and dev server error)

### PR DESCRIPTION
**Pull request by jhildenbiddle**
## **Summary**

Fix a collection of legacy browsers bugs.

**NOTE:** Obviously, legacy browsers are not a huge priority but the fact remains that Docsify v4.x does officially support legacy browsers like IE11. These non-breaking fixes address issues I bump into most often when testing IE11 from time to time.

## **What kind of change does this PR introduce?**

* Fix CSS styles in legacy browsers without support for CSS custom properties
* Fix presentation of search field
* Fix an error on the website caused by linking to the non-transpired version of `docsify-plugin-carbon`
* Fix dev server error in IE (BrowserSync snippet not compatible / now loaded conditionally)
* Refactor `/docs` site conditional JS loading (simplification)

**Additional changes**

* Fix ESLint errors found in `/docs/index.html`
* Fix `.gitignore` paths that blocked committing changes to `/src/themes`

**Screenshots - Before**

CleanShot 2022-02-02 at 23 48 53@2x

CleanShot 2022-02-02 at 23 49 53@2x

**Screenshots - After**

CleanShot 2022-02-02 at 23 43 39@2x

CleanShot 2022-02-03 at 11 00 01@2x

## **For any code change,**

- [ ] Related documentation has been updated if needed
- [ ] Related tests have been updated or tests have been added

## **Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

## **Tested in the following browsers:**

- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge
- [x] IE

### CODE REVIEW

**Review by Koooooo-7 — DISMISSED** [review]

**Comment by sy-records** [comment]
There seems to be some problems here.

image

**Comment by jhildenbiddle** [comment]
@sy-records 

> There seems to be some problems here.

Fixed.

FYI, the issue you highlighted in your screenshot wasn't introduced with this PR. The search field has always been like that in IE11. It was easy enough to fix, so we may as well while we're addressing (what will hopefully be the last) IE issues.

File: src/core/init/lifecycle.js [hunk 1]
```diff
@@ -36,14 +36,27 @@ export function Lifecycle(Base) {
...
-            hookFn(data, result => {
-              data = result;
-              step(index + 1);
-            });
+            try {
+              hookFn(data, result => {
+                data = result;
+              });
+            } catch (err) {
+              console.error(errTitle, err);
```
Comment (trusktr) [inline src/core/init/lifecycle.js:47]: Can we keep this in the separate PR?

**Review by trusktr — CHANGES_REQUESTED** [review]
Introduces the same change as in https://github.com/docsifyjs/docsify/pull/1742. Needs to be split? (or maybe this one should merge onto that one)

File: src/core/init/lifecycle.js (hunk 1 above)
Comment (jhildenbiddle) [reply src/core/init/lifecycle.js:47]: Yes. This change should not have been merged into this PR. Sorry about that. Good catch. I'll remove it.

Comment (jhildenbiddle) [reply src/core/init/lifecycle.js]: @trusktr -- Reverted. Thanks again for the catch.

File: docs/index.html [hunk 2]
```diff
@@ -55,7 +55,7 @@
 
   <body>
     <div id="app">Loading ...</div>
-    <script src="//cdn.jsdelivr.net/npm/docsify-plugin-carbon@1/index.js"></script>
+    <script src="//cdn.jsdelivr.net/npm/docsify-plugin-carbon@1"></script>
```
Comment (sy-records) [inline docs/index.html:58]: Do you need to update index.html?

https://github.com/docsifyjs/docsify/blob/533193fd2d27f7edae051ea4de80514c09545248/index.html#L26

Comment (jhildenbiddle) [reply docs/index.html:58]: Yep. :)

Done.

**Review by sy-records — APPROVED** [review]
LGTM.

**Review by Koooooo-7 — APPROVED** [review]
LGTM.

**Comment by jhildenbiddle** [comment]
Thanks @sy-records and @Koooooo-7! 🙏
